### PR TITLE
GitHub desktop port

### PR DESCRIPTION
**Pull request by usersxx**
###### Type(s)

- [x] enhancement

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by i0ntempest — CHANGES_REQUESTED** [review]
App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

**Comment by i0ntempest** [comment]
`applications_dir` is set by MacPorts and you're not supposed to overwrite it. Also why are you copying the whole `${worksrcpath}`?

**Comment by usersxx** [comment]
> App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

Removed `violate_mtree`, `extract.dir` in lines 37 and 34. Added `${applications_dir}`, `${destroot}${applications_dir}` in lines 34 and 36

**Comment by usersxx** [comment]
The worksrcdir is set to `github desktop.app`

**Comment by i0ntempest** [comment]
My bad for missing that, but variables set by MacPorts shouldn't be changed unless absolutely necessary. The right thing to do here is extract the app to the default `${worksrcpath}` and copy `${worksrcpath}${app_name}` to `${destroot}${applications_dir}`. By changing `${applications_dir}` you're still violating mtree.

**Comment by i0ntempest** [comment]
Right, so I got your portfile and tried it. You indeed have to overwrite `worksrcdir`, but you should not dictate where to install an app for everyone in your portfile. `${applications_dir}` defaults to `/Applications/MacPorts/` and can be changed by a user in a config file. So `copy ${worksrcpath} ${destroot}${applications_dir}` will copy the app just fine.

Other things:
Version should be 2.9.6-9196a1ae, right now you're splitting it;
Change `extract.suffix      .zip` to `use_zip yes` so extraction actually works;
Your file name should be `Portfile` with a capital P. macOS filesystems can be case sensitive and they are on MacPorts buildbots. Pay attention to other places where you might mess up cases as well (Github vs GitHub etc.);
Maintainer field should look like this: `{@name domain:username}`, so for you maybe {@usersxx gmail.com:usersxx}.

File: devel/Github-Desktop/Portfile
Comment (i0ntempest) [inline devel/Github-Desktop/Portfile:12]: change this line to `maintainers        {@usersxx microsoft.com:usersword123} openmaintainer`

File: devel/Github-Desktop/Portfile
Comment (i0ntempest) [inline devel/Github-Desktop/Portfile:38]: Add a few whitespaces to format this line like other lines

**Comment by usersxx** [comment]
> App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.

You should approve changes

**Comment by i0ntempest** [comment]
> > App bundles installed by MacPorts should be installed to `${applications_dir}` so you don't need `violate_mtree`. `extract.dir` should not be touched, rather you should write a `destroot` block to copy the app to `${destroot}${applications_dir}`. See qBittorrent portfile for an example.
> 
> You should approve changes so the changes requested section should go away

Not yet, after you've done with the changes you need to squash your commits into one and change your commit message to follow our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages). Basically:
`port_name: summary of changes`

**Comment by usersxx** [comment]
Ok

**Comment by i0ntempest** [comment]
Right, I did the format change for you and the checks are running now. You can now squash all changes and I'll merge if no other members have suggestions in a few days

**Comment by usersxx** [comment]
> Right, I did the format change for you and the checks are running now. You can now squash all changes and I'll merge if no other members have suggestions in a few days

Ok

**Review by reneeotten — CHANGES_REQUESTED** [review]
this PR should not be merged in my opinion. We want to build packages from source and not just copy binaries from a website - in cases like this just install the thing directly from upstream.

If you people really, really want to do this then you will need to make sure to disable automatic updates through the app since now MacPorts will have to take care of this, amongst probably other things.

**Comment by i0ntempest** [comment]
> this PR should not be merged in my opinion. We want to build packages from source and not just copy binaries from a website - in cases like this just install the thing directly from upstream.
> 
> If you people really, really want to do this then you will need to make sure to disable automatic updates through the app since now MacPorts will have to take care of this, amongst probably other things.

Actually I've thought about that. This app is written in typescript, is there any port currently builds typescript to binary? If so then maybe it's doable... But it's probably too hard for a first time contributer to do though.

**Comment by usersxx** [comment]
Actually there is a repository for github desktop. [https://github.com/desktop/desktop](url)

File: devel/Github-Desktop/Portfile
Comment (ryandesign) [inline devel/Github-Desktop/Portfile:5]: Capitalize GitHub correctly.
```suggestion
name                GitHub-Desktop
```

**Review by ryandesign — COMMENTED** [review]
I agree the port should build from source, if possible.

If not, I left some comments about improvements that can be made.

The port fails to activate for me on Catalina due to [this MacPorts bug](https://trac.macports.org/ticket/60749)—did it activate for you? On what macOS version? (You didn't fill in that part of the pull request template.)

```
Error: Failed to activate GitHub-Desktop: Image error: Source file /opt/local/var/macports/software/GitHub-Desktop/mpextract5jq2fMoo/Applications/MacPorts/GitHub Desktop.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Resources/._MainMenu.nib does not appear to exist (cannot lstat it).  Unable to activate port GitHub-Desktop.
```

File: devel/Github-Desktop/Portfile
Comment (ryandesign) [reply devel/Github-Desktop/Portfile:5]: The name of the directory that the Portfile is in should be updated to match this capitalization.

File: devel/Github-Desktop/Portfile
Comment (usersxx) [reply devel/Github-Desktop/Portfile:14]: The pull request name is GitHub but the directory of the Portfile is macports-ports/devel/Github-Desktop/Portfile

**Comment by usersxx** [comment]
I think you should see this file - https://github.com/desktop/desktop/blob/development/package.json

**Pull request closed without merging**
